### PR TITLE
Pal 390 resource service springboot cacheable

### PR DESCRIPTION
**Pull request by pd104923**
Applied changes to HadoopResourceService made within common

**Requires PR in common to be merged before a successful build**

### CODE REVIEW

File: hadoop-resource/src/test/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline hadoop-resource/src/test/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceServiceTest.java]: update copyright

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 2]
```diff
@@ -152,81 +147,76 @@ private static Configuration createConfig(final Map<String, String> conf) {
...
+        LOGGER.debug("Invoking getResourcesByResource for resource: {}", resource);
+        return getResourcesById(resource.getId());
     }
 
     @Override
-    public CompletableFuture<Map<LeafResource, ConnectionDetail>> getResourcesById(final GetResourcesByIdRequest request) {
-        requireNonNull(request, "request");
-        LOGGER.debug("Invoking getResourcesById request: {}", request);
-        final String resourceId = request.getResourceId();
+    public Map<LeafResource, ConnectionDetail> getResourcesById(final String str) {
```
Comment (developer01189998819991197253) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java]: rename "str" to something more descriptive

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 3]
```diff
@@ -152,81 +147,76 @@ private static Configuration createConfig(final Map<String, String> conf) {
...
         }
-        return getFutureMappings(resourceId, ignore -> true);
+        return getMappings(str, ignore -> true);
     }
 
     @Override
-    public CompletableFuture<Map<LeafResource, ConnectionDetail>> getResourcesByType(final GetResourcesByTypeRequest request) {
-        requireNonNull(request, "request");
-        LOGGER.debug("Invoking getResourcesByType request: {}", request);
+    public Map<LeafResource, ConnectionDetail> getResourcesByType(final String str) {
```
Comment (developer01189998819991197253) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java]: rename "str" to something more descriptive

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java [hunk 4]
```diff
@@ -152,81 +147,76 @@ private static Configuration createConfig(final Map<String, String> conf) {
...
-        return getFutureMappings(pathString, predicate);
+        final Predicate<ResourceDetails> predicate = detail -> str.equals(detail.getType());
+        return getMappings(pathString, predicate);
     }
 
     @Override
-    public CompletableFuture<Map<LeafResource, ConnectionDetail>> getResourcesBySerialisedFormat(final GetResourcesBySerialisedFormatRequest request) {
-        requireNonNull(request, "request");
-        LOGGER.debug("Invoking getResourcesBySerialisedFormat request: {}", request);
+    public Map<LeafResource, ConnectionDetail> getResourcesBySerialisedFormat(final String str) {
```
Comment (developer01189998819991197253) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java]: rename "str" to something more descriptive

File: hadoop-resource/src/test/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceServiceTest.java (hunk 1 above)
Comment (pd104923) [reply hadoop-resource/src/test/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceServiceTest.java]: Updated

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java (hunk 2 above)
Comment (pd104923) [reply hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java]: Renamed

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java (hunk 3 above)
Comment (pd104923) [reply hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java]: Renamed

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java (hunk 4 above)
Comment (pd104923) [reply hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/service/HadoopResourceService.java]: Renamed

**Comment by ac74475** [comment]
closing while design issues are resolved that have come up from this work

**Pull request closed without merging**